src/frontend/src/optimizer/plan_node/logical_share.rs

```diff
@@ -118,6 +120,14 @@ impl ColPrunable for LogicalShare {
     }
 }
 
+impl ExprRewritable for LogicalShare {
+    fn rewrite_exprs(&self, _r: &mut dyn ExprRewriter) -> PlanRef {
+        unimplemented!(
+            "call rewrite_exprs of the PlanRef instead of calling directly on LogicalShare"
+        )
+    }
+}
```

Comment:
Just like the comment tells us. We should handle `LogicalShare` in `rewrite_exprs` of  `PlanRef `.

Reply:
Apparently, it is handled properly with clone_with_inputs. Let me confirm.

Reply:
I've decided to treat this as a noop instead.